### PR TITLE
Enable csi snapshotter in stable overlay

### LINKED ISSUE #457: Update deployments to include new snapshot controller (for v1beta1 APIs), start running tests, and update docs
Looks like some new snapshot APIs went Beta in k8s `1.17`. The corresponding `external-snapshotter` is also available and tested with the PD Driver. However, the stable deployment in this repo still does not contain the snapshotter, tests are not run by default, and the documentation for snapshotting may be out of date.

### PR DESCRIPTION
**Pull request by saikat-royc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
Enable csi snapshotter sidecar  (beta) in stable overlays.
Add necessary csi snapshotter specific manifests to base and update the diffs in the release-staging-head, release-staging-rc, stable, alpha overlays.
**Which issue(s) this PR fixes**:

Fixes #457 

**Special notes for your reviewer**:
This is part 1 of the Pull Request.
Subsequent PR will include documentation changes and bump the pd csi driver version.
**Does this PR introduce a user-facing change?**:

```release-note
Enable csi snapshotter side car (beta) for PD CSI driver.
```

### CODE REVIEW

File: deploy/kubernetes/base/controller.yaml [hunk 1]
```diff
@@ -48,6 +48,15 @@ spec:
           volumeMounts:
             - name: socket-dir
               mountPath: /csi
+        - name: csi-snapshotter
+          imagePullPolicy: Always
```
Comment (msau42) [inline deploy/kubernetes/base/controller.yaml:52]: remove the imagePullPolicy here. Usually it's unnecessary to have to always pull a new image, unless you're reusing the same tag, which is generally only applicable to dev scenarios.

File: deploy/kubernetes/base/kustomization.yaml [hunk 2]
```diff
@@ -6,3 +6,4 @@ resources:
 - node.yaml
 - controller.yaml
 - setup-cluster.yaml
+- rbac_add_snapshotter.yaml
```
Comment (msau42) [inline deploy/kubernetes/base/kustomization.yaml:9]: Can you append this to "setup-cluster.yaml" instead? This is where all the rbacs usually live.

File: deploy/kubernetes/overlays/alpha/kustomization.yaml [hunk 3]
```diff
@@ -2,17 +2,15 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 bases:
 - ../stable
-patches:
-- controller_add_snapshotter.yaml
+images:
+- name: gke.gcr.io/csi-snapshotter
```
Comment (msau42) [inline deploy/kubernetes/overlays/alpha/kustomization.yaml:6]: I think we can remove support for the alpha snapshot feature.

File: test/run-k8s-integration.sh [hunk 4]
```diff
@@ -44,7 +44,7 @@ else
   base_cmd="${base_cmd} --gce-region=${gce_region}"
 fi
 
-if [[ "$overlay_name" =~ .*"gke-release-staging".* ]]; then
+if [[ "$overlay_name" =~ .*"gke-release-staging".* ]] || [[ "$overlay_name" = "stable" ]]; then
```
Comment (msau42) [inline test/run-k8s-integration.sh:47]: We can remove this check entirely since snapshotter should be in all overlays now

Comment (saikat-royc) [reply test/run-k8s-integration.sh:47]: This is assuming that no integration tests are run against alpha?

File: deploy/kubernetes/base/kustomization.yaml (hunk 2 above)
Comment (saikat-royc) [reply deploy/kubernetes/base/kustomization.yaml:9]: Ack

File: deploy/kubernetes/base/controller.yaml (hunk 1 above)
Comment (saikat-royc) [reply deploy/kubernetes/base/controller.yaml:52]: Ack

**Comment by msau42** [comment]
Can you also update the README, as well as the examples?

**Comment by msau42** [comment]
(and user guides)

File: deploy/kubernetes/overlays/alpha/kustomization.yaml (hunk 3 above)
Comment (saikat-royc) [reply deploy/kubernetes/overlays/alpha/kustomization.yaml:6]: Can you please clarify @msau42 ? Do you mean use beta snapshot side car here for the alpha overlay?

**Comment by saikat-royc** [comment]
> Can you also update the README, as well as the examples?

Sure. Will do as a separate PR to update all necessary docs.

File: deploy/kubernetes/overlays/alpha/kustomization.yaml (hunk 3 above)
Comment (saikat-royc) [reply deploy/kubernetes/overlays/alpha/kustomization.yaml:6]: Ack

**Comment by msau42** [comment]
/lgtm
/approve

snapshot tests were run and passed in the pull job

**Pull request merged**